frame/composable-traits/src/lending/math.rs

```diff
@@ -281,54 +275,44 @@ impl DynamicPIDControllerModel {
 		utilization_ratio: FixedU128,
 	) -> Result<Rate, ArithmeticError> {
 		// compute error term `et = uo - ut`
-		let et: i128 = self.uo.into_inner().try_into().unwrap_or(0i128) -
+		let et: i128 = self.target_utilization.into_inner().try_into().unwrap_or(0i128) -
 			utilization_ratio.into_inner().try_into().unwrap_or(0i128);
 		let et: FixedI128 = FixedI128::from_inner(et);
 		// compute proportional term `pt = kp * et`
-		let pt = self.kp.checked_mul(&et).ok_or(ArithmeticError::Overflow)?;
+		let pt = self.proportional_parameter.checked_mul(&et).ok_or(ArithmeticError::Overflow)?;
 		//compute integral term `it = it_1 + ki * et`
 		let it = self
-			.it_1
-			.checked_add(&self.ki.checked_mul(&et).ok_or(ArithmeticError::Overflow)?)
+			.previous_integral_term
+			.checked_add(&self.integral_parameter.checked_mul(&et).ok_or(ArithmeticError::Overflow)?)
 			.ok_or(ArithmeticError::Overflow)?;
-		self.it_1 = it;
+		self.previous_integral_term = it;
 		// compute derivative term `dt = kd * (et - et_1)`
-		let dt = self.kd.checked_mul(&(et - self.et_1)).ok_or(ArithmeticError::Overflow)?;
-		self.et_1 = et;
+		let dt = self.derivative_parameter.checked_mul(&(et - self.previous_error_value)).ok_or(ArithmeticError::Overflow)?;
+		self.previous_error_value = et;
 
 		// compute u(t), control value `ut = pt + it + dt`
 		let ut = pt + it + dt;
 		// update interest_rate `ir = ir_t_1 + ut`
-		if ut.is_negative() {
```

Comment:
IIRC `et` can be negative and based on that other values like `ut` can be negative @dzmitry-lahoda  right?